src/strategies/conv-finance/index.ts

```diff
@@ -0,0 +1,347 @@
+              stakingPoolsV2ContractAbi,
+              [
+                [stakingPool.address, 'poolInfos', [poolInfo.poolId]], // Get pool token
+                ...addresses.map((userAddress: any) => [
+                  stakingPool.address,
+                  'userData',
+                  [poolInfo.poolId, userAddress]
+                ]),
+                ...addresses.map((userAddress: any) => [
+                  stakingPool.address,
+                  'getReward',
+                  [poolInfo.poolId, userAddress, poolInfo.rewarderIdx]
+                ])
+              ],
+              { blockTag }
+            )
+          );
+        });
+      } else {
+        stakingPool.pools.forEach((poolInfo: STAKING_POOL_INFO) => {
+          stakingPoolsMultiCalls.push(
+            multicall(
+              network,
+              provider,
+              stakingPoolsV1ContractAbi,
+              [
+                [stakingPool.address, 'poolInfos', [poolInfo.poolId]], // Get pool token
+                ...addresses.map((userAddress: any) => [
+                  stakingPool.address,
+                  'userData',
+                  [poolInfo.poolId, userAddress]
+                ]),
+                ...addresses.map((userAddress: any) => [
+                  stakingPool.address,
+                  'getReward',
+                  [poolInfo.poolId, userAddress]
+                ])
+              ],
+              { blockTag }
+            )
+          );
+        });
+      }
+    });
+  }
+
+  const stakingPoolRewarderMultiCalls: any[] = [];
+
+  options.rewarder.forEach((rewarder: REWARDER) => {
+    if(rewarder.poolIds.length <= 5){
+      //  Staking pool rewarder version:
+      // 1: Old rewarder
+      // 2. New rewarder
+      if (rewarder.version === '2') {
+        rewarder.poolIds.forEach((id: number) => {
+          stakingPoolRewarderMultiCalls.push(
+            multicall(
+              network,
+              provider,
+              rewarderV2ContractAbi,
+              [
+                ...addresses.map((userAddress: any) => [
+                  rewarder.address,
+                  'calculateTotalReward',
+                  [userAddress, id]
+                ])
+              ],
+              { blockTag }
+            )
+          );
+        });
+      } else {
+        rewarder.poolIds.forEach((id: number) => {
+          stakingPoolRewarderMultiCalls.push(
+            multicall(
+              network,
+              provider,
+              rewarderV1ContractAbi,
+              [
+                ...addresses.map((userAddress: any) => [
+                  rewarder.address,
+                  'vestingSchedules',
+                  [userAddress, id]
+                ])
+              ],
+              { blockTag }
+            )
+          );
+        });
+      }
+    }
+  });
+
+  let res = await Promise.all([
+    ...lpMultiCalls,
+    ...stakingPoolsMultiCalls,
+    ...stakingPoolRewarderMultiCalls
+  ]);
```

Comment:
Hello @eric-convergence This definitely crossing number of requests sent from a strategy, there is a limit of 5 requests you can send from a strategy, https://github.com/snapshot-labs/snapshot-strategies#checklist-for-adding-a-new-strategy


Can you try if there is any alternate way for this, thanks :)

Reply:
I have merge all contract calls into 3 multi-call request, and it should be independent of the pool and rewarder size.